### ISSUE TITLE
Crash setting up pod

### ISSUE THREAD

**Issue by jsoref** [linkerd/linkerd2#5493]
### What is the issue?
Afaict, #5036 introduced a null pointer deref near 
https://github.com/linkerd/linkerd2/blob/2ff70d4c7da2af2dbe868d470bbacb9de0275607/pkg/inject/inject.go#L569

### How can it be reproduced?
We're just trying to roll out new pods (using ArgoCD) that happen to have linkerd sidecars (via standard injection)

### Logs, error output, etc

```
Error creating: Internal error occurred: failed calling webhook "linkerd-proxy-injector.linkerd.io": Post https://linkerd-proxy-injector.linkerd.svc:443/?timeout=30s: EOF
```
```
time="2021-01-07T21:25:33Z" level=info msg="received pod/nginx-ingress-default-backend-dfbb7b9c-"
2021/01/07 21:25:33 http: panic serving 127.0.0.1:59656: runtime error: invalid memory address or nil pointer
dereference
goroutine 410 [running]:
net/http.(*conn).serve.func1(0xc000018320)
        /usr/local/go/src/net/http/server.go:1772 +0x139
panic(0x18ed5c0, 0x2c142e0)
        /usr/local/go/src/runtime/panic.go:975 +0x3e3
github.com/linkerd/linkerd2/pkg/inject.(*ResourceConfig).injectPodSpec(0xc001082400, 0xc00000b2c0)
        /linkerd-build/pkg/inject/inject.go:569 +0x47d
github.com/linkerd/linkerd2/pkg/inject.(*ResourceConfig).GetPatch(0xc001082400, 0xc0005d1801, 0x1b62a2f, 0x15,
 0xc0005d19b8, 0xc001555f80, 0x23)
        /linkerd-build/pkg/inject/inject.go:270 +0x7fa
github.com/linkerd/linkerd2/controller/proxy-injector.Inject(0x1e52b20, 0xc0011164c0, 0xc000379380, 0xc0005088
20, 0x1e4bfe0, 0xc000747a00, 0x1, 0xafd, 0x600)
        /linkerd-build/controller/proxy-injector/webhook.go:99 +0x10ae
github.com/linkerd/linkerd2/controller/webhook.(*Server).processReq(0xc0007f4930, 0x1e52b20, 0xc0011164c0, 0xc
000207000, 0xafd, 0xe00, 0x0)
        /linkerd-build/controller/webhook/server.go:122 +0x359
github.com/linkerd/linkerd2/controller/webhook.(*Server).serve(0xc0007f4930, 0x1e4c960, 0xc00013b5e0, 0xc00182
e300)
        /linkerd-build/controller/webhook/server.go:93 +0x84
net/http.HandlerFunc.ServeHTTP(0xc0007eefb0, 0x1e4c960, 0xc00013b5e0, 0xc00182e300)
        /usr/local/go/src/net/http/server.go:2012 +0x44
net/http.serverHandler.ServeHTTP(0xc00079a380, 0x1e4c960, 0xc00013b5e0, 0xc00182e300)
        /usr/local/go/src/net/http/server.go:2807 +0xa3
net/http.(*conn).serve(0xc000018320, 0x1e52b20, 0xc001116100)
        /usr/local/go/src/net/http/server.go:1895 +0x86c
created by net/http.(*Server).Serve
        /usr/local/go/src/net/http/server.go:2933 +0x35c
```

#### `linkerd check` output

```text
your output here ...
```

### Environment

- Kubernetes Version: Server Version: version.Info{Major:"1", Minor:"17+", GitVersion:"v1.17.13-gke.2001", GitCommit:"00c919adfe4adf308bcd7c02838f2a1b60482f02", GitTreeState:"clean", BuildDate:"2020-11-06T18:24:02Z", GoVersion:"go1.13.15b4", Compiler:"gc", Platform:"linux/amd64"}
- Cluster Environment: GKE
- Host OS:
- Linkerd version: 2.9.1
ghcr.io/linkerd/controller:stable-2.9.1ghcr.io/linkerd/proxy-init:v1.3.8 and ghcr.io/linkerd/proxy:stable-2.9.1

### Possible solution
I'm pretty sure a null check would help

**Comment by adleong** [comment]
This has been fixed by https://github.com/linkerd/linkerd2/pull/5269, and the fix is available in edge-20.11.5 and later.

**Comment by jsoref** [comment]
Thanks, we moved to edge-21.1.1 and this part of things is happy (lots of other things are still in the air, but...)
